packages/teams-js/src/internal/communication.ts

```diff
@@ -766,20 +770,25 @@ function handleChildMessage(evt: DOMMessageEvent): void {
       // @ts-ignore
       sendMessageResponseToChild(message.id, message.uuid, Array.isArray(result) ? result : [result]);
     } else {
-      // No handler, proxy to parent
-      sendMessageToParent(
-        getApiVersionTag(ApiVersionNumber.V_2, ApiName.Tasks_StartTask),
-        message.func,
-        message.args,
-        (...args: any[]): void => {
-          if (Communication.childWindow) {
-            const isPartialResponse = args.pop();
-            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-            // @ts-ignore
-            sendMessageResponseToChild(message.id, message.uuid, args, isPartialResponse);
-          }
-        },
-      );
+      if (GlobalVars.allowMessageProxy || GlobalVars.webAuthWindowOpen) {
```

Comment:
Both of these new GlobalVars start out with a value of false when the app is initialized. `allowMessageProxy` is only true if an undocumented function is called. `webAuthWindowOpen` is only true if `authentication.authenticate` is called from an app running in a browser, and it is changed back to false as soon as that authentication window is closed.